orcid_hub/authcontroller.py

```diff
@@ -824,7 +824,8 @@ def orcid_login(invitation_token=None):
             user_id=None, method="GET", url=orcid_authenticate_url, state=state)
         oac.save()
 
-        return redirect(orcid_authenticate_url)
+        return render_template(
+            "orcidLogoutAndCallback.html", orcid_base_url=ORCID_BASE_URL, callback_url=orcid_authenticate_url)
```

Comment:
This would be case when you wouldn't need to introduce a template. you can return HTM content directly:
```
return f"""<!DOCTYPE html>
<html>
<head>
	<script src="https://code.jquery.com/jquery-2.1.4.js"></script>
    <script>
        $.ajax({
            url: "{orcid_base_url}/userStatus.json?logUserOut=true",
            dataType: 'jsonp',
            success: function(result,status,xhr) {
              window.location='{{callback_url|safe}}'
            },
            error: function (xhr, status, error) {
              alert(status);
            }
        });
    </script>
</head>
<body>
</body>
</html>"""
```

Reply:
After discussion with Rad, we notice that there is no clean way to do logout, since ORCID needs an ajax call for logout from there side, and we have to redirect the user to give permission as soon as we do the force logout. so will keep these changes till we come up with a clean way to do logout from ORCID.